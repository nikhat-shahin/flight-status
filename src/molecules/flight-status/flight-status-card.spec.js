import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from "enzyme-adapter-react-16";
import { createSerializer } from 'enzyme-to-json';
import toJson from 'enzyme-to-json';
import FlightStatus from './flight-status-card';

configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

it('should render Page Component correctly', () => {
  const component = shallow(
    < FlightStatus airlineDesignator = {
      `EK`
    }
    flightNumber = {
      `0007`
    }
    flightId = {
      `2022031600007DXB`
    }
    flightDate = {
      `2022-03-16`
    }
    flightRoute = {
      [{
        "legNumber": "1",
        "originActualAirportCode": "Dubai (DXB)",
        "destinationActualAirportCode": "London (LHR)",
        "originPlannedAirportCode": "Dubai (DXB)",
        "destinationPlannedAirportCode": "London (LHR)",
        "statusCode": "ARVD",
        "flightPosition": 100,
        "totalTravelDuration": "08:00",
        "isIrregular": "false",
        "departureTime": {
          "schedule": "2022-03-17T03:10:00Z",
          "estimated": "2022-03-17T03:10:00Z",
          "actual": "2022-03-17T03:15:00Z"
        },
        "arrivalTime": {
          "schedule": "2022-03-17T07:10:00Z",
          "estimated": "2022-03-17T06:53:00Z",
          "actual": "2022-03-17T07:08:00Z"
        },
        "operationalUpdate": {
          "lastUpdated": "2022-01-20T04:00:55Z"
        },
        "departureTerminal": "Terminal 3",
        "arrivalTerminal": "Terminal 3",
        "flightOutageType": 0
      }]
    }
    booked = {
      true
    }
    buttonId = {
      1
    }
    />,
  );
  expect(toJson(component)).toMatchSnapshot();
});