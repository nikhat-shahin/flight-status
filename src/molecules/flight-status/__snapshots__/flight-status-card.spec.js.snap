// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Page Component correctly 1`] = `
<div
  className="flight-status__grey-container"
>
  <Container
    fluid={false}
  >
    <Row
      className="flight-status__row"
    >
      <Col
        className="flight-status__col"
      >
        <div
          className="flight-status__time-container"
        >
          <p
            className="flight-status__airport"
          >
            Dubai (DXB)
          </p>
          <p
            className="flight-status__state"
          >
            Departed:
          </p>
          <p
            className="flight-status__time"
          >
            03:15
          </p>
          <p
            className="flight-status__date"
          >
            Thu 17 Mar
          </p>
          <p
            className="flight-status__scheduled-time"
          >
            Scheduled Departure:
            03:10
          </p>
        </div>
      </Col>
      <Col
        className="flight-status__col"
      >
        <img
          alt="express"
          className="flight-status__arrow-img"
          src="arrow.png"
        />
      </Col>
      <Col
        className="flight-status__col"
      >
        <div
          className="flight-status__time-container"
        >
          <p
            className="flight-status__airport"
          >
            London (LHR)
          </p>
          <p
            className="flight-status__state"
          >
            Arrived:
          </p>
          <p
            className="flight-status__time"
          >
            07:08
          </p>
          <p
            className="flight-status__date"
          >
            Thu 17 Mar
          </p>
          <p
            className="flight-status__scheduled-time"
          >
            Scheduled Arrival:
            07:10
          </p>
        </div>
      </Col>
      <Col
        className="flight-status__status"
      >
        <div
          className="flight-status__flightstatus flight-status__ARVD__color"
        >
          Arrived
        </div>
        <p
          className="flight-status__flight-number-text"
        >
          <img
            alt="express"
            className="flight-status__flight-number-tail"
            src="flighttail.svg"
          />
           
          EK 0007
        </p>
        <img
          alt="express"
          className="flight-status__flight-number-triangle"
          src="triangleedge.png"
        />
      </Col>
    </Row>
  </Container>
</div>
`;
